### DISCUSSION TITLE
Remove support for `##...` Java types

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#7873] [Ideas]
The proposal is that Java types must always be imported, i.e:

```scala
import java.util.Comparator

mod Adaptor {

    /// ...
    pub def comparator(_: Proxy[a]): Comparator \ IO with Order[a] =
```

is OK, but this will no longer be allowed:

```scala
mod Adaptor {

    /// ...
    pub def comparator(_: Proxy[a]): ##java.util.Comparator \ IO with Order[a] =
```

**Comment by JonathanStarup** [comment]
This feels a bit weird with the interop improvement. In X months the example will be "look at this nice code with this large ugly type import block at the start"

**Reply by JonathanStarup** [reply]
Java imports are also a bit simple - If I need the java `List` I have to import that type directly, so now I can't use normal `List`.

I could write `import java....{List => JavaList}` but its seems preferable to me to use a qualified name or a partially qualified name

**Reply by magnus-madsen** [reply]
You can do a named import:

```
import java.util.{List => JList}
```

**Reply by magnus-madsen** [reply]
You only have to import types, so the situation will be pretty much like Java, Scala, etc.

**Comment by jaschdoc** [comment]
Should we close this?
